Adds tests for isnan().

danbrown-amd opened the pull request:
Test values taken from the isinf() tests; others may be relevant.

danbrown-amd committed:
Adds tests for isnan().

farzonl reviewed (approved):
test LGTM. 

Will just want to confirm test passes across vendors before merging.
If you see issues feel free to file issues and add vendor specific xfails for now.

danbrown-amd commented:
On the last test run, 8 tests passed, 9 failed with errors in previously existing tests, and 6 did not run to completion.  I rebased and updated on my end, with the only change in the added tests being that I removed some obsolete comment lines.

bogner committed:
Merge branch 'main' into isnan

bogner commented on test/Feature/HLSLLib/isnan.16.test:
```diff
@@ -0,0 +1,65 @@
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [1, 1, 1]
```
```suggestion
```

bogner commented on test/Feature/HLSLLib/isnan.32.test:
```diff
@@ -0,0 +1,60 @@
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [1, 1, 1]
```
```suggestion
```

bogner committed:
Remove DispatchSize

it's been moved to DispatchParameters and matches the default anyway

Co-authored-by: Justin Bogner <mail@justinbogner.com>
